src/Symfony/Component/RateLimiter/SlidingWindowLimiter.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+use Symfony\Component\Lock\LockInterface;
+use Symfony\Component\Lock\NoLock;
+use Symfony\Component\RateLimiter\Storage\StorageInterface;
+use Symfony\Component\RateLimiter\Util\TimeUtil;
+
+/**
+ * The sliding window algorithm will look at your last window and the current one.
+ * It is good algorithm to reduce bursts.
+ *
+ * Example:
+ * If your limit is 10. Last time window we did 8 hits. We are currently 25% into
+ * the current window. We have made 3 hits in current window.
+ * That means our sliding window hit count is (75% * 8) + 3 = 9.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * @experimental in 5.2
+ */
+final class SlidingWindowLimiter implements LimiterInterface
+{
+    private $id;
```

Comment:
Thanks for the docs above, would love to get some here too 😎

Reply:
Sure!